Add filtering and sorting to rules configuration

klinki opened the pull request:
- filtering by `source` or `target` technology (for now only text based without version support, I'll create separate PR for it)
- sorting by:
  - name
  - count of rules
  - date created
  - last modified

jsight commented:
I see a couple of issues that I'm sure would produce some complaints:

- The tags aren't quite aligned. See: http://imgur.com/a/Uzm4v - The "target: eap" is slightly higher than the filter label
- I don't think that we should expose "created date" and "Last modified date". They aren't clearly exposed in the UI and it isn't clear to me what the user could do with them.

Otherwise, this looks really nice.

klinki commented:
- fixed tags alignment
- removed created date/last modified date
- reduced spacing between header and toolbar

[![full-page-screenshot.png](https://s30.postimg.org/6bpc0ulup/full-page-screenshot.png)](https://postimg.org/image/7dzije4nx/)

klinki commented:
retest

jsight commented:
@klinki - I am still seeing errors with the production build of this PR. If you fix those, I think that this will be ready to merge.
